test/python/test_pool.py

```diff
@@ -0,0 +1,164 @@
+
+import pytest
+from pytest_bdd import given, scenario, then, when, parsers
+from common.command import run_cmd
+import mayastor_pb2 as pb
+
+
+@scenario('features/pool.feature', 'creating a pool using disk with invalid block size')
+def test_creating_a_pool_using_disk_with_invalid_block_size():
+    'Creating a pool using disk with invalid block size.'
+
+
+@scenario('features/pool.feature', 'creating a pool with a name that already exists')
+def test_creating_a_pool_with_a_name_that_already_exists():
+    'Creating a pool with a name that already exists.'
+
+
+@scenario('features/pool.feature', 'creating a pool with an AIO disk')
+def test_creating_a_pool_with_an_aio_disk():
+    'Creating a pool with an AIO disk.'
+
+
+@scenario('features/pool.feature', 'creating a pool with multiple disks')
+def test_creating_a_pool_with_multiple_disks():
+    'Creating a pool with multiple disks.'
+
+
+@scenario('features/pool.feature', 'destroying a pool')
+def test_destroying_a_pool():
+    'Destroying a pool.'
+
+
+@scenario('features/pool.feature', 'destroying a pool that does not exist')
+def test_destroying_a_pool_that_does_not_exist():
+    'Destroying a pool that does not exist.'
+
+
+@scenario('features/pool.feature', 'listing pools')
+def test_listing_pools():
+    'Listing pools.'
+
+
+@pytest.fixture(scope="module")
+def mayastor_instance(mayastor_mod):
+    yield mayastor_mod['ms0']
+
+
+@pytest.fixture(scope="module")
+def find_pool(mayastor_instance):
+    def lookup(name):
+        for pool in mayastor_instance.ms.ListPools(pb.Null()).pools:
+            if pool.name == name:
+                return pool
+        return None
+    yield lookup
+
+
+@pytest.fixture
+def create_image_file():
+    filename = '/tmp/ms0-aio.img'
+    run_cmd(f"rm -f '{filename}'", True)
+    run_cmd(f"truncate -s 64M '{filename}'", True)
+    yield filename
+    run_cmd(f"rm -f '{filename}'", True)
+
+
+@pytest.fixture
```

Comment:
I see what you are trying to do here, but why not yield in create pool and then have it destroyed? Fixtures are nice but my (short) experience is that they can make things complicated and surprising when not needed.

Reply:
create_pool is a (given) step function, and is definitely not a fixture. Placing a yield here causes an error (even if nothing happens subsequently. I'm not sure if this is a bug as such or by design.
In any case create_pool is not supposed to destroy the pool as its job is to create the pool in order to test the DestroyPool command.